sandbox/OpenIddict.Sandbox.AspNet.Server/Startup.cs

```diff
@@ -272,6 +298,117 @@ await manager.CreateAsync(new OpenIddictApplicationDescriptor
+                        Permissions.Scopes.Roles,
+                        Permissions.Prefixes.Scope + "demo_api"
+                    },
+                    Requirements =
+                    {
+                        Requirements.Features.ProofKeyForCodeExchange
+                    }
+                });
+            }
+
+            if (await manager.FindByClientIdAsync("avalonia") is null)
+            {
+                await manager.CreateAsync(new OpenIddictApplicationDescriptor
+                {
+                    ApplicationType = ApplicationTypes.Native,
+                    ClientId = "avalonia",
+                    ClientType = ClientTypes.Public,
+                    ConsentType = ConsentTypes.Systematic,
+                    DisplayName = "Avalonia client application",
+                    DisplayNames =
+                    {
+                        [CultureInfo.GetCultureInfo("fr-FR")] = "Application cliente avalonia"
+                    },
+                    PostLogoutRedirectUris =
+                    {
+                        new Uri("com.openiddict.sandbox.avalonia.client:/callback/logout/local")
+                    },
+                    RedirectUris =
+                    {
+                        new Uri("com.openiddict.sandbox.avalonia.client:/callback/login/local")
+                    },
+                    Permissions =
+                    {
+                        Permissions.Endpoints.Authorization,
+                        Permissions.Endpoints.Logout,
+                        Permissions.Endpoints.Token,
+                        Permissions.GrantTypes.AuthorizationCode,
+                        Permissions.GrantTypes.RefreshToken,
+                        Permissions.ResponseTypes.Code,
+                        Permissions.Scopes.Email,
+                        Permissions.Scopes.Profile,
+                        Permissions.Scopes.Roles,
+                        Permissions.Prefixes.Scope + "demo_api"
+                    },
+                    Requirements =
+                    {
+                        Requirements.Features.ProofKeyForCodeExchange
+                    }
+                });
+            }
         }).GetAwaiter().GetResult();
     }
 }
+
+internal static class ServiceCollectionExtensions
+{
+    /// <summary>
+    ///     Note: Do not use the following keys in production!
+    ///     there just added so we can authenticate from Android (Maui/Avalonia), iOS (Maui/Avalonia) and Macos, since on these platforms the following does not work
+    ///     
+    ///       options.AddDevelopmentEncryptionCertificate()
+    ///         .AddDevelopmentSigningCertificate();
+    ///     
+    ///     this is because
+    ///     1. Adding an X509 certificate to the system store is not supported (throws on iOS and Android)
+    ///     2. and even if so, it would be a different machine - therefore different store - and client and server would not use the same certificate!
```

Comment:
Huh no, the crypto material used by the server is completely unrelated to the crypto material used by a client (otherwise, it would offer no security at all 🤣). Revert all that stuff, please.

Reply:
hoopsiedaysie - I guess that will go ito the guide too then :-D